auth/src/main/java/com/firebase/ui/auth/viewmodel/idp/ProviderHandlerBase.java

```diff
@@ -0,0 +1,23 @@
+package com.firebase.ui.auth.viewmodel.idp;
+
+import android.app.Application;
+import android.content.Intent;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.annotation.RestrictTo;
+
+import com.firebase.ui.auth.IdpResponse;
+import com.firebase.ui.auth.viewmodel.ViewModelBase;
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+public abstract class ProviderHandlerBase<T extends ProviderHandlerParamsBase> extends ViewModelBase<T> {
```

Comment:
I truly can't get my head around the `Provider`, `Providerbase`, `ProviderHandler`, `ProviderHandlerbase`, `ProviderHandlerParams`, and `ProviderHandlerParamsBase` classes.  The names are so similar and they all seem to be dependent on each other. 

I would love to find a way to simplify them.  I wish I had a more constructive idea and not just this comment.

Reply:
Yeah, I was trying to make it so a single `LiveData` could be used for all providers, but clearly, that's just made things confusing. I'll fix that and have the picker activity just subscribe to each provider's individual `LiveData`. Then I think things will quickly become clearer. 👍